src/components/search-filter-input/SearchFilterInput.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState, KeyboardEvent } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import AppTextField from '~/components/app-text-field/AppTextField'
+import { TextFieldProps } from '@mui/material/TextField'
+import SearchIcon from '@mui/icons-material/Search'
+import Button from '@mui/material/Button'
+import Box from '@mui/material/Box'
+import ClearIcon from '@mui/icons-material/Clear'
+import IconButton from '@mui/material/IconButton'
+
+import { SizeEnum, VariantEnum } from '~/types'
+
+import { styles } from '~/components/search-filter-input/SearchFilterInput.styles'
+
+interface SearchFilterInputProps {
+  updateFilter: (value: string) => void
+  textFieldProps: TextFieldProps
+}
+
+const SearchFilterInput = ({
+  updateFilter,
+  textFieldProps
+}: SearchFilterInputProps) => {
+  const [search, setSearch] = useState<string>('')
+  const { t } = useTranslation()
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearch(event.target.value)
+  }
+
+  const onSearch = () => {
+    updateFilter(search)
+  }
+
+  const onClear = () => {
+    setSearch('')
+    updateFilter('')
+  }
+
+  const onEnterPress = (event: KeyboardEvent<HTMLInputElement>) => {
+    event.key === 'Enter' && onSearch()
+  }
+
+  const labelStyle = {
+    ...styles.inputLabel,
+    visibility: search && 'hidden'
+  }
+
+  const clearIconVisibility = { visibility: search ? 'visible' : 'hidden' }
```

Comment:
Need to change on:
  visibility: searchInput && VisibilityEnum.Hidden
  }
  const clearIconVisibility = {visibility: search ? VisibilityEnum.Visible : VisibilityEnum.Hidden